### PR TITLE
Update references and use HTTPS when available

### PR DESCRIPTION
**Pull request by carusogabriel**
Only `http://codepad.viper-7.com` I couldn't solve.

### CODE REVIEW

File: _posts/2011-12-12-How-big-are-PHP-arrays-really-Hint-BIG.markdown [hunk 1]
```diff
@@ -281,17 +281,17 @@ bytes. That's 56 bytes per element and thus much less than the 144 bytes per ele
 uses. This is because a fixed array doesn't need the bucket structure: So it only requires one zval
 (48 bytes) and one pointer (8 bytes) for each element, giving us the observed 56 bytes.
 
-  [3]: http://en.wikipedia.org/wiki/Union_%28computer_science%29
-  [4]: http://lxr.php.net/opengrok/xref/PHP_5_4/Zend/zend.h#307
+  [3]: https://en.wikipedia.org/wiki/Union_%28computer_science%29
+  [4]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend.h#307
```
Comment (nikic) [inline _posts/2011-12-12-How-big-are-PHP-arrays-really-Hint-BIG.markdown]: This is `#322` in 5.6.

File: _posts/2011-12-12-How-big-are-PHP-arrays-really-Hint-BIG.markdown [hunk 2]
```diff
@@ -281,17 +281,17 @@ bytes. That's 56 bytes per element and thus much less than the 144 bytes per ele
...
   [5]: http://blog.golemon.com/2007/01/youre-being-lied-to.html
-  [6]: http://lxr.php.net/opengrok/xref/PHP_5_4/Zend/zend_hash.h#54
+  [6]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_hash.h#54
```
Comment (nikic) [inline _posts/2011-12-12-How-big-are-PHP-arrays-really-Hint-BIG.markdown]: This is `#55` in 5.6.

File: _posts/2011-12-12-How-big-are-PHP-arrays-really-Hint-BIG.markdown [hunk 3]
```diff
@@ -281,17 +281,17 @@ bytes. That's 56 bytes per element and thus much less than the 144 bytes per ele
...
   [7]: http://schlueters.de/blog/
   [8]: http://www.tyrael.hu/
-  [9]: http://php.net/manual/en/features.gc.collecting-cycles.php
-  [10]: http://lxr.php.net/xref/PHP_5_4/Zend/zend.h#318
-  [11]: http://lxr.php.net/opengrok/xref/PHP_5_4/Zend/zend_gc.h#zval_gc_info
-  [12]: http://php.net/manual/en/internals2.memory.php
+  [9]: https://php.net/manual/en/features.gc.collecting-cycles.php
+  [10]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend.h#318
```
Comment (nikic) [inline _posts/2011-12-12-How-big-are-PHP-arrays-really-Hint-BIG.markdown]: This is `#334` in 5.6.

File: _posts/2011-12-12-How-big-are-PHP-arrays-really-Hint-BIG.markdown [hunk 4]
```diff
@@ -281,17 +281,17 @@ bytes. That's 56 bytes per element and thus much less than the 144 bytes per ele
...
+  [11]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_gc.h#zval_gc_info
```
Comment (nikic) [inline _posts/2011-12-12-How-big-are-PHP-arrays-really-Hint-BIG.markdown]: `#91` gives a better link here.

File: _posts/2011-12-12-How-big-are-PHP-arrays-really-Hint-BIG.markdown [hunk 5]
```diff
@@ -281,17 +281,17 @@ bytes. That's 56 bytes per element and thus much less than the 144 bytes per ele
...
+  [12]: https://php.net/manual/en/internals2.memory.php
   [13]: http://g.oswego.edu/dl/html/malloc.html
-  [14]: http://lxr.php.net/xref/PHP_5_4/Zend/zend_alloc.c#336
-  [15]: http://php.net/SplFixedArray
-  [16]: https://nikic.github.io/2014/12/22/PHPs-new-hashtable-implementation.html
\ No newline at end of file
+  [14]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_alloc.c#336
```
Comment (nikic) [inline _posts/2011-12-12-How-big-are-PHP-arrays-really-Hint-BIG.markdown]: This is `#366` in 5.6.

File: _posts/2012-03-16-Understanding-PHPs-internal-function-definitions.markdown [hunk 6]
```diff
@@ -244,14 +244,14 @@ In the next part
...
-  [9]: http://lxr.php.net/opengrok/xref/PHP_5_4/Zend/zend_builtin_functions.c#478
-  [10]: http://lxr.php.net/opengrok/search?q=SplFixedArray%3A%3AgetSize&project=PHP_5_4
-  [11]: http://blog.ircmaxell.com/
\ No newline at end of file
+  [1]: https://blog.ircmaxell.com/2012/03/phps-source-code-for-php-developers.html
+  [2]: https://lxr.room11.org/xref/php-src@5.6
+  [3]: https://lxr.room11.org/search?project=php-src@5.6&q=strpos
+  [4]: https://lxr.room11.org/search?project=php-src@5.6&q="PHP_FUNCTION+strpos"
+  [5]: https://lxr.room11.org/xref/php-src@5.6/ext/standard/php_string.h#48
+  [6]: https://lxr.room11.org/xref/php-src@5.6/ext/standard/string.c#1789
```
Comment (nikic) [inline _posts/2012-03-16-Understanding-PHPs-internal-function-definitions.markdown]: This is `#1827` in 5.6.

File: _posts/2012-03-16-Understanding-PHPs-internal-function-definitions.markdown [hunk 7]
```diff
@@ -244,14 +244,14 @@ In the next part
...
+  [7]: https://php.net/sprintf
+  [8]: https://lxr.room11.org/search?project=php-src@5.6&q="ZEND_FUNCTION+strlen"
+  [9]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_builtin_functions.c#478
```
Comment (nikic) [inline _posts/2012-03-16-Understanding-PHPs-internal-function-definitions.markdown]: This is `#481` in 5.6.

File: _posts/2012-03-28-Understanding-PHPs-internal-array-implementation.markdown [hunk 8]
```diff
@@ -297,15 +297,15 @@ In the next part
...
- [8]: http://lxr.php.net/xref/PHP_5_4/Zend/zend_hash.h#98
- [9]: http://lxr.php.net/opengrok/xref/PHP_TRUNK/Zend/zend_API.h#353
- [10]: http://de.php.net/array_fill_keys
- [11]: http://nikic.github.com/2012/03/16/Understanding-PHPs-internal-function-definitions.html
- [12]: http://blog.ircmaxell.com/
\ No newline at end of file
+ [1]: https://blog.ircmaxell.com/2012/03/phps-source-code-for-php-developers.html
+ [2]: https://nikic.github.com/2012/03/16/Understanding-PHPs-internal-function-definitions.html
+ [3]: https://blog.ircmaxell.com/2012/03/phps-source-code-for-php-developers_21.html
+ [4]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_hash.h#261
```
Comment (nikic) [inline _posts/2012-03-28-Understanding-PHPs-internal-array-implementation.markdown]: This is `#269` in 5.6.

File: _posts/2012-03-28-Understanding-PHPs-internal-array-implementation.markdown [hunk 9]
```diff
@@ -297,15 +297,15 @@ In the next part
...
+ [5]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_hash.h#66
```
Comment (nikic) [inline _posts/2012-03-28-Understanding-PHPs-internal-array-implementation.markdown]: This is `#67` in 5.6.

File: _posts/2012-03-28-Understanding-PHPs-internal-array-implementation.markdown [hunk 10]
```diff
@@ -297,15 +297,15 @@ In the next part
...
+ [6]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_hash.h#54
```
Comment (nikic) [inline _posts/2012-03-28-Understanding-PHPs-internal-array-implementation.markdown]: This is `#55` in 5.6.

File: _posts/2012-03-28-Understanding-PHPs-internal-array-implementation.markdown [hunk 11]
```diff
@@ -297,15 +297,15 @@ In the next part
...
+ [7]: https://lxr.room11.org/xref/php-src@5.6/ext/standard/array.c#1583
```
Comment (nikic) [inline _posts/2012-03-28-Understanding-PHPs-internal-array-implementation.markdown]: This is `#1551` in 5.6.

File: _posts/2012-03-28-Understanding-PHPs-internal-array-implementation.markdown [hunk 12]
```diff
@@ -297,15 +297,15 @@ In the next part
...
+ [8]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_hash.h#98
+ [9]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_API.h#353
```
Comment (nikic) [inline _posts/2012-03-28-Understanding-PHPs-internal-array-implementation.markdown]: This is `#364` in 5.6.

File: _posts/2012-07-27-How-to-add-new-syntactic-features-to-PHP.markdown [hunk 13]
```diff
@@ -599,16 +599,16 @@ Oh, and one last thing: The `in` operator here was just an example. I don't plan
...
-  [vm_gen]: http://lxr.php.net/xref/PHP_TRUNK/Zend/zend_vm_execute.h
-  [xref]: http://lxr.php.net
-  [internals_list]: http://php.net/mailing-lists.php
+  [scanner_def]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_language_scanner.l
+  [scanner_gen]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_language_scanner.c
+  [parser_def]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_language_parser.y
+  [compiler]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_compile.c
+  [compiler_header]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_compile.h
+  [tokenizer]: https://php.net/tokenizer
+  [zend_op]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_compile.h#106
```
Comment (nikic) [inline _posts/2012-07-27-How-to-add-new-syntactic-features-to-PHP.markdown]: This is `#111` in 5.6.

File: _posts/2012-07-27-How-to-add-new-syntactic-features-to-PHP.markdown [hunk 14]
```diff
@@ -599,16 +599,16 @@ Oh, and one last thing: The `in` operator here was just an example. I don't plan
...
+  [znode_op]: https://lxr.room11.org/xref/php-src@5.6/Zend/zend_compile.h#74
```
Comment (nikic) [inline _posts/2012-07-27-How-to-add-new-syntactic-features-to-PHP.markdown]: This is `#78` in 5.6.

**Comment by carusogabriel** [comment]
Updated those lines references :+1:

**Pull request merged**